plugins/module_utils/azure_rm_common.py

```diff
@@ -876,6 +876,12 @@ def get_mgmt_svc_client(self, client_type, base_url=None, api_version=None, supp
             # most things are resource_manager, don't make everyone specify
             base_url = self.azure_auth._cloud_environment.endpoints.resource_manager
 
+        # https://github.com/Azure/msrestazure-for-python/pull/169
+        # China's base_url doesn't end in a trailing slash, though others do,
+        # and we need a trailing slash when generating credential_scopes below.
+        if base_url[-1] != "/":
```

Comment:
Does the string end with a '/'? Isn't it?

Reply:
kinlly ping!

Reply:
Right -- if the string doesn't end with a slash, then we add one. I'm realizing, now that I look back at this, that python has `str.endswith()`. I can rewrite this commit to use that instead, if you prefer.

Reply:
Ok, Thanks for your reply!

Reply:
> Right -- if the string doesn't end with a slash, then we add one. I'm realizing, now that I look back at this, that python has `str.endswith()`. I can rewrite this commit to use that instead, if you prefer.

This comments, Please!

Reply:
Done! Thanks for your patience; I wanted to make sure I'd tested everything, to not accidentally re-introduce some regression.